common/achievement/types.ts

```diff
@@ -7,7 +7,8 @@ export type Metric = {
 
 export type EventValue = number | string | boolean;
 
-export type FormulaContext = {
+// match, pool, openMatchRequestCount, screening, firstMatchRequest
+export type Context = {
```

Comment:
If you want to reuse the existing actions, why do we need a "Context" specific to achievements? Can't we use the notification context ("action context")?